irohad/maintenance/metrics.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "maintenance/metrics.hpp"
+
+#include <prometheus/counter.h>
+#include <prometheus/exposer.h>
+#include <prometheus/registry.h>
+
+#include <array>
+#include <chrono>
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <thread>
+#include <future>
+#include <regex>
+
+auto maintenance_metrics_init(std::string const& listen_addr)
+->std::shared_ptr<prometheus::Registry>
```

Comment:
You don't use compile-time return type calculation.
``` c++
std::shared_ptr<prometheus::Registry>
maintenance_metrics_init(std::string const& listen_addr)
```

Reply:
Here I use only new syntax. But I changed it for consistancy